### DISCUSSION TITLE
login page not working

### DISCUSSION THREAD

**Discussion by tauseedzaman** [filamentphp/filament#11107] [Help]
### Package

filament/filament

### Package Version

V3.2

### Laravel Version

V10.8

### Livewire Version

V3.1

### PHP Version

V8.2.12

### Problem description

after a fresh install i visited `/admin/login` and see this loading bar on the button automatically apperis..

![image](https://github.com/filamentphp/filament/assets/64689921/6c4d220a-bb25-4b54-a759-cc52d4d9fea6)

i tried to login with correct credentials... it didn't work then i tried with wrong credentials it didn't worked again but didn't showed any error also...

plz advise

### Expected behavior

i have used filament before many times.. but haven't seen this behavior now is it a bug in this version or i am doing something wrong...?

### Steps to reproduce

just installed filament in my project... no customization are made.

### Reproduction repository

https://github.com/tauseedzaman/inventory-management-system

### Relevant log output

```shell
no info available in logs.. logs are clear.. no error, no warnings...
```

**Comment by danharrin** [comment]
Please start from a fresh Laravel project, currently I cannot install your repository due to unrelated NPM issues.

And I do not think this is a problem with Filament, so I'm going to convert this to a discussion. You might get faster help on Discord for diagnosis.

**Reply by jhonoryza** [reply]
@danharrin can filament using livewire config `inject_assets => false` ?

**Reply by fliix-cloud** [reply]
Same problem still exists in Filament 5.x - as soon as Livewire has "inject_assets => false" Filament breaks. Can't find anything in the docs.

**Comment by tauseedzaman** [comment]
its an open source project i am by my self contributing to... that project don't have admin panel i am tiring to use filament... but i am unable to use it in there...

**Reply by leandrocfe** [reply]
change in config/livewire.php

```php
'inject_assets' => true
```

**Reply by CharrafiMed** [reply]
thanks a lot bro

**Reply by hasan-aftab** [reply]
Thanks a lot @leandrocfe brother

But I am not able to access admin/login route

**Comment by mwagena** [comment]
Had the same issue and this was the fix!

> change in config/livewire.php
> 
> ```
> 'inject_assets' => true
> ```

**Comment by AFSDARIF** [comment]
I got the same problem for this case, have tried adding

'inject_assets' => true in config/livewire.php

but still can't log in until now

**Comment by leandrocfe** [comment]
> I got the same problem for this case, have tried adding
> 
> 'inject_assets' => true in config/livewire.php
> 
> but still can't log in until now

Fresh install?

**Reply by divdax** [reply]
same for me. on local i can login. on my staging i get a 405. filament assets are loaded but what i noticed is that the password is plaintext by default and i cannot switch on clicking the eye. The login itself should not send a post request to admin/login. instead it should be a livewire request?

**Reply by irfanandratama** [reply]
> same for me. on local i can login. on my staging i get a 405. filament assets are loaded but what i noticed is that the password is plaintext by default and i cannot switch on clicking the eye. The login itself should not send a post request to admin/login. instead it should be a livewire request?

I am experiencing same issue, but manage to "fix" it. It was mention at https://github.com/filamentphp/filament/discussions/11481
Try to run php artisan vendor:publish --force --tag=livewire:assets
Or add the following code: 

use Illuminate\Support\Facades\Route;
use Livewire;
`Livewire::setUpdateRoute(function ($handle) {
    return Route::post('/custom/livewire/update', $handle);
});`
to your AppServiceProvider.php

It getting rid of 405 but in my place still can't login, the sign in button just load but not redirecting to dashboard.

**Comment by divdax** [comment]
It seems to be a Livewire related thing. It's working with LIvewire v3.4.6:

```
composer require livewire/livewire:v3.4.6 -W
```

**Reply by Fludem** [reply]
do you have repository to reproduce

**Reply by divdax** [reply]
no, just tested in my project so far. On my local env everything works even with the newst Livewire version. It was just the deployed project.

**Reply by divdax** [reply]
@danharrin we're not alone with this problem. See also discord: https://discord.com/channels/883083792112300104/1214941718365999194

**Comment by Fludem** [comment]
@divdax @danharrin 

### First Test
Just to be clear I've just performed a fresh install of filament with the latest version of livewire and I can't replicate this issue.

Here's the repo: [Fresh Install Repo](https://github.com/Fludem/Filament-Livewire-Login-Bug-11107/)

Composer lock file showing latest livewire version is available here: [composer.lock](
https://github.com/Fludem/Filament-Livewire-Login-Bug-11107/blob/b949f6fcb92bbbf30dd6984d7b0f38db0d088ea0/composer.lock#L3320)

### Observations

Livewire recently shipped with an Alpine version that broke some things.

Livewire 3.4.8 includes an Alpine version without these issues:

> This release contains an updated version of Alpine (3.13.7) that reverts a few recent breaking changes. Here's the notable Alpine release notes: https://github.com/alpinejs/alpine/releases/tag/v3.13.7

You can see the release here: [Livewire 3.4.8 Release](https://github.com/livewire/livewire/releases/tag/v3.4.8)

**Reply by Fludem** [reply]
@divdax can you run these:

```bash
php artisan optimize:clear
php artisan filament:upgrade
```

and try logging in again, do it in incognito of possible.

**Reply by Fludem** [reply]
Please see updates above @divdax and check your livewire version.

**Reply by divdax** [reply]
I reinstalled Livewire 3.4.8 and run `optimize:clear` and `filament:upgrade` with the same result. Form is submitting a post request to /admin/login which doesn't exist and the password field is in plaintext and i can't toggle. Just to be clear: This only happens on my staging server. On my local machine everything is working fine.

// edit: Also no console errors on the login page. 

composer packages i use:
```
    "repositories": [
        {
            "type": "git",
            "url": "https://github.com/divdax/LaravelShoppingcart.git"
        }
    ],
    "require": {
        "php": "^8.3",
        "artesaos/seotools": "^1.1",
        "bumbummen99/shoppingcart": "dev-master",
        "filament/filament": "^v3.2.44",
        "filament/spatie-laravel-media-library-plugin": "^3.0-stable",
        "filament/spatie-laravel-tags-plugin": "^3.0-stable",
        "guzzlehttp/guzzle": "^7.2",
        "laravel/framework": "^10",
        "laravel/horizon": "^5.14",
        "laravel/pulse": "^1.0@beta",
        "laravel/sanctum": "^3.0",
        "laravel/tinker": "^2.7",
        "league/flysystem-aws-s3-v3": "^3.0",
        "livewire/livewire": "^3.0",
        "lorisleiva/laravel-actions": "^2.5",
        "openai-php/client": "*",
        "secondnetwork/blade-tabler-icons": "^2.41",
        "shetabit/visitor": "^4.2",
        "shuvroroy/filament-spatie-laravel-backup": "^2.1",
        "spatie/browsershot": "^3.57",
        "spatie/laravel-backup": "^8.1",
        "spatie/laravel-db-snapshots": "^2.5",
        "spatie/laravel-markdown": "^2.5",
        "spatie/laravel-sitemap": "^6.2",
        "spatie/laravel-sluggable": "^3.4",
        "spatie/laravel-tags": "^4.3",
        "stripe/stripe-php": "^10.8"
    },
    "require-dev": {
        "barryvdh/laravel-debugbar": "^3.8",
        "deployer/deployer": "^7.1",
        "fakerphp/faker": "^1.9.1",
        "laravel-lang/lang": "^12.12",
        "laravel/breeze": "^1.19",
        "laravel/pint": "^1.0",
        "laravel/sail": "^1.0.1",
        "mockery/mockery": "^1.4.4",
        "nunomaduro/collision": "^7.2",
        "pestphp/pest": "^2.0",
        "pestphp/pest-plugin-faker": "^2.0",
        "pestphp/pest-plugin-laravel": "^2.0",
        "pestphp/pest-plugin-livewire": "^2.0",
        "spatie/laravel-ignition": "^2.0",
        "spatie/laravel-ray": "^1.32",
        "worksome/request-factories": "^3.1"
    },
```

**Reply by leandrocfe** [reply]
You should create a repository on github to reproduce this issue

**Reply by adityaa-codes** [reply]
@divdax hey, how did you solve this issue?

**Comment by danharrin** [comment]
Users of this thread who are @ing me - you've identified its a LW issue, can you let me know what you want me to do personally? As far as I can see there isn't action on the Filament side we should be taking? Can you clarify?

**Reply by Fludem** [reply]
I just @'d you so you didn't waste your time as I proved was LW issue

**Reply by danharrin** [reply]
@divdax?

**Reply by divdax** [reply]
Yes, it's a Livewire issue. When it impacts several Filament users, i thought you should be aware. Sorry for stealing your time.

**Reply by danharrin** [reply]
Just FYI, I'm not experiencing issues logging in with the latest Livewire version. I would strongly suggest checking your browser caching and any asset caching you may have through something like Cloudflare.

**Comment by maryam-mardani** [comment]
1. Create a Login class in this directory: app/Filament/Pages/Auth
2. extends from \Filament\Pages\Auth\Login 
3. Change the method "throwFailureValidationException" , default is data.mobile, change it to data.email


class Login extends \Filament\Pages\Auth\Login{

    protected function throwFailureValidationException(): never
    {
        throw ValidationException::withMessages([
            'data.email' => __('filament-panels::pages/auth/login.messages.failed'),
        ]);
    }

}

4. set custom page as your panel login in Providers/Filament/AdminPanelProvider:



use App\Filament\Pages\Auth\Login;
 public function panel(Panel $panel): Panel
    {
        return $panel
            ->default()
            ->login(Login::class)
.
.
.
}

**Comment by softwaretechniker** [comment]
A potential cause with a nearly fresh setup: When you removed cookie / session relevant middleware from Laravels "web" middleware group.
This causes the same error behavior - even when session middlewares are still configured for Filament's panel.

For anyone wondering why someone doesn't like to start with a session cookie without having a login:
This is could be a legal issue in some countries, especially when you want to avoid unnecessary "Do you accept our cookies?" prompts.

**Comment by anoshiri** [comment]
I answered this in another thread https://github.com/filamentphp/filament/discussions/11481#discussioncomment-10301278

**Comment by uclitapasillas** [comment]
php artisan vendor:publish --force --tag=livewire:assets

Eso lo soluciona

**Reply by danharrin** [reply]
You should not do this, it will result in outdated assets in the future

**Comment by ersin-demirtas** [comment]
I have the exact same issue, I did some digging into Livewire the SupportAutoInjectedAssets tries to injectAssets but this is done after the request handled event. Where the content is already gzipped so it can't detect the </html> with the gzip response and it simply skips. 

Not sure if this is any use to any one but check if your response is encoded its probably a middleware. In my case I was using the GzipEncodeResponse::class from the ErlandMuchasaj\LaravelGzip. This may be different for you, if you are encoding into different one. 

I hope this helps.

**Comment by maks-oleksyuk** [comment]
I ran into a similar issue when running Laravel on a path (`doma.in/laravel`) instead of the root domain(`doma.in`). My custom Livewire routes didn’t have the `web` middleware, so `$guard->check()` returned false even though the user was logged in. Adding `->middleware(['web'])` to the Livewire `setUpdateRoute` and `setScriptRoute` fixed the problem for me.

example of my code:

```php
Livewire::setUpdateRoute(fn ($handler) => Route::post($updateUrl, $handler)->name(ltrim((string) $prefix, '/').'.livewire.update')->middleware(['web']));
Livewire::setScriptRoute(fn ($handler) => Route::get($scriptUrl, $handler)->name(ltrim((string) $prefix, '/').'.livewire.scripts')->middleware(['web']));
```

Maybe this will save someone some time :octocat:
